controllers/controller/workspacerouting/predictes.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package workspacerouting
+
+import (
+	"errors"
+
+	controllerv1alpha1 "github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var routingPredicates = predicate.Funcs{
+	CreateFunc: func(ev event.CreateEvent) bool {
+		obj, ok := ev.Object.(*controllerv1alpha1.WorkspaceRouting)
+		if !ok {
+			return true
```

Comment:
This could be come a reconcile issue -- we could apply an annotation saying `routingClass: basic` or something, but we'd still have to trigger reconciles for resources with no annotation. Otherwise, if the annotation is removed for some reason, it would no longer trigger reconciles on modification, potentially breaking routing. We could make some sort of agreement to add a "managed-by" annotation to externally-created routing objects, but I don't know if it makes sense to handle that now.

Reply:
That should not block that PR